src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
-// write code here
+const headers = document.querySelectorAll('th');
+const tBody = document.querySelector('tBody');
+const rows = document.querySelectorAll('tBody tr');
+
+const array = [...rows];
+const copyArray = [...array];
+
+headers.forEach((header) => {
+  header.addEventListener('click', function (element) {
+    const index = element.target.cellIndex;
+
+    sortBy(copyArray, index);
+  });
+});
+
+function sortBy(arrayOf, index) {
+  arrayOf.sort((td1, td2) => {
+    if (index === 0 || index === 1) {
+      return td1.children[index].innerHTML.localeCompare(
+        td2.children[index].innerHTML,
+      );
+    } else {
+      return (
+        Number(
+          td1.children[index].innerHTML.replace(',', '').replace('$', ''),
+        ) -
+        Number(td2.children[index].innerHTML.replace(',', '').replace('$', ''))
+      );
+    }
+  });
+
+  arrayOf.forEach((tr) => {
+    tBody.appendChild(tr);
+  });
+}
```

Comment:
This `sortBy` function is doing too much. It's sorting the array, manipulating the DOM, and also has some special logic for different types of data. Consider breaking it down into smaller functions. For example, one function could be responsible for sorting the array based on the type of data, and another for updating the DOM.